client/custom_flow.go

```diff
@@ -0,0 +1,103 @@
+package client
+
+type CustomFlowCreatePayload struct {
+	Name                 string           `json:"name"`
+	Repository           string           `json:"repository"`
+	Path                 string           `json:"path,omitempty"`
+	Revision             string           `json:"revision,omitempty"`
+	TokenId              string           `json:"tokenId,omitempty"`
+	SshKeys              []TemplateSshKey `json:"sshKeys,omitempty"`
+	GithubInstallationId int              `json:"githubInstallationId,omitempty"`
+	BitbucketClientKey   string           `json:"bitbucketClientKey,omitempty"`
+	IsBitbucketServer    bool             `json:"isBitbucketServer"`
+	IsGitlabEnterprise   bool             `json:"isGitLabEnterprise"`
+	IsGithubEnterprise   bool             `json:"isGitHubEnterprise"`
+	IsGitLab             bool             `json:"isGitLab"`
+	IsAzureDevOps        bool             `json:"isAzureDevOps" tfschema:"is_azure_devops"`
+	IsTerragruntRunAll   bool             `json:"isTerragruntRunAll"`
+}
```

Comment:
Is it possible to add some composition with the already existing attribute of the template resource?

Reply:
Yes! It's possible. It makes more sense to me.

Embedded composition in golang is explicit. See this for an explanation:
https://blog.dlow.me/programming/golang/struct-embedding/#golang-struct-embedding

This means that I will have to make changes in all the code locations where Template is initialized
E.g.:
```
	template := client.Template{
		Id:               "id0",
		Name:             "template",
...

```

It's not an issue - just additional (many) cosmetic changes in the PR.
Let me know if I should proceed. Another option is to apply it as a follow-up PR.

Reply:
Yeah it's probably not worth it...